apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/admin/http/HttpTokenStandardMetadataHandler.scala

```diff
@@ -60,12 +65,37 @@ class HttpTokenStandardMetadataHandler(
       }
   }
 
-  private def lookupTotalSupply()(implicit ec: ExecutionContext, tc: TraceContext) = (
+  private def lookupTotalSupplyByLatestRound()(implicit ec: ExecutionContext, tc: TraceContext) =
     for {
       (latestRoundNr, effectiveAt) <- OptionT(store.lookupRoundOfLatestData())
       totalSupply <- OptionT.liftF(store.getTotalAmuletBalance(latestRoundNr))
     } yield (totalSupply, effectiveAt)
-  ).value
+
+  private def lookupTotalSupplyByLatestAcsSnapshot()(implicit tc: TraceContext) = {
+    for {
+      latestSnapshot <- OptionT(
+        acsSnapshotStore.lookupSnapshotBefore(acsSnapshotStore.currentMigrationId, clock.now)
+      )
+      unlocked <- OptionT.fromOption[Future](latestSnapshot.unlockedAmuletBalance)
+      locked <- OptionT.fromOption[Future](latestSnapshot.lockedAmuletBalance)
+    } yield (locked + unlocked) -> latestSnapshot.snapshotRecordTime.toInstant
+  }
+
+  private def lookupTotalSupply()(implicit tc: TraceContext) = {
+    for {
+      noHoldingFeesOnTransfers <- packageVersionSupport.noHoldingFeesOnTransfers(
+        store.key.dsoParty,
+        clock.now,
+      )
+      deductHoldingFees = !noHoldingFeesOnTransfers.supported
+      result <-
+        if (deductHoldingFees) {
+          lookupTotalSupplyByLatestRound().value
+        } else {
+          lookupTotalSupplyByLatestAcsSnapshot().orElse(lookupTotalSupplyByLatestRound()).value
```

Comment:
hm, do we have any way to report back the time of this value? Is there some metadata field that we can put it into?

Reply:
We'll need it at least in the scan UI. I guess the token standard endpoints also didn't mention the round before so maybe not a hard blocker for those but seems kind of annoying to not have it.